boot/modeenv.go

```diff
@@ -328,50 +328,35 @@ func (m *Modeenv) WriteTo(rootdir string) error {
 	return nil
 }
 
-// modeenvModelForSealing is a helper type that implements
-// github.com/snapcore/secboot.SnapModel interface with additional methods used
-// internally in the boot package.
-type modeenvModelForSealing interface {
-	BrandID() string
-	SignKeyID() string
-	Model() string
-	Grade() asserts.ModelGrade
-	Series() string
-	uniqueID() string
-}
-
-// dummy to verify interface match
-var _ secboot.ModelForSealing = (modeenvModelForSealing)(nil)
-
-// modeenvModelForSealing is a helper type that implements
-// github.com/snapcore/secboot.SnapModel and modeenvModelForSealing interfaces.
-type modeenvModelForSealingImpl struct {
+// modelForSealing is a helper type that implements
+// github.com/snapcore/secboot.SnapModel interface.
+type modelForSealing struct {
 	brandID        string
 	model          string
 	grade          asserts.ModelGrade
 	modelSignKeyID string
 }
 
 // dummy to verify interface match
-var _ secboot.ModelForSealing = (*modeenvModelForSealingImpl)(nil)
+var _ secboot.ModelForSealing = (*modelForSealing)(nil)
 
-func (m *modeenvModelForSealingImpl) BrandID() string           { return m.brandID }
-func (m *modeenvModelForSealingImpl) SignKeyID() string         { return m.modelSignKeyID }
-func (m *modeenvModelForSealingImpl) Model() string             { return m.model }
-func (m *modeenvModelForSealingImpl) Grade() asserts.ModelGrade { return m.grade }
-func (m *modeenvModelForSealingImpl) Series() string            { return release.Series }
+func (m *modelForSealing) BrandID() string           { return m.brandID }
+func (m *modelForSealing) SignKeyID() string         { return m.modelSignKeyID }
+func (m *modelForSealing) Model() string             { return m.model }
+func (m *modelForSealing) Grade() asserts.ModelGrade { return m.grade }
+func (m *modelForSealing) Series() string            { return release.Series }
 
 // uniqueID returns a unique ID which can be used as a map index of the
 // underlying model.
-func (m *modeenvModelForSealingImpl) uniqueID() string {
+func (m *modelForSealing) uniqueID() string {
```

Comment:
this could be a function modelUniqueID that takes a secboot.ModelForSealing and returns 

"brand-id/model,grade,model-sign-key"